### ISSUE TITLE
Window icon is wrong (generic "X" icon)

### ISSUE THREAD

**Issue by geekley** [flathub/com.github.PintaProject.Pinta#19]
In flatpak, Pinta icon in windows is wrong (it shows the generic "X" icon), and I can only fix it by using Kubuntu Window Rules and setting the "pinta" window class to the desktop file name "com.github.PintaProject.Pinta.desktop".
Something must be missing in the flatpak packaging, or in the upstream code.

The icon shows normally in the start menu and taskbar, it's only wrong in the windows (both normal and dialog windows).

---
Operating System: Kubuntu 21.04 amd64
KDE Plasma Version: 5.21.4
KDE Frameworks Version: 5.80.0
Qt Version: 5.15.2
Kernel Version: 5.11.0-38-generic
Graphics Platform: X11

**Comment by darkdragon-001** [comment]
I have no idea what could be wrong. It works in Ubuntu Gnome. If you find out what is wrong, I am happy to accept a PR :wink:

**Comment by darkdragon-001** [comment]
@geekley Can you reproduce this issue with the latest version? Can you reproduce it also with the non-Flatpak one?

**Comment by geekley** [comment]
It happens in:
v2.0.1, flatpak
v2.0.1, snap

Doesn't happen in:
v1.6-2, apt from ubuntu impish/universe amd64

I didn't find a non-sandboxed v2.0.1 to test (I think the tar.gz and zip are just sources). I don't know if this is a bug in pinta that was fixed in apt by maintainers, or if it's a bug in both snap and flatpak packaging of pinta, or if it's a bug in pinta that appears in both because of the sandboxing.
But I can still fix it like I said above, with Window Rules.
Maybe this should be reported upstream (I can't remember if I already did that, I think I didn't).

**Comment by geekley** [comment]
How weird. Now I can't reproduce it anymore ... All I did was installing/uninstalling the snap and apt versions of pinta ...
I thought it might have to do with `gnome-icon-theme` which was installed by apt but didn't uninstall when I purged pinta. But even after I purged that too, the icon is still appearing properly when I start flatpak pinta. I would try restarting (icon cache?), but I'm kinda busy now :sweat_smile: Anyways, it MIGHT be related to that.
